nixos/modules/services/taler/module.nix

```diff
@@ -0,0 +1,108 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+
+let
+  cfg = config.services.taler;
+  settingsFormat = pkgs.formats.ini { };
+in
+
+{
+  # TODO turn this into a generic taler-like service thingy?
+  options.services.taler = {
+    enable = lib.mkEnableOption "the GNU Taler system" // lib.mkOption { internal = true; };
+    includes = lib.mkOption {
+      type = lib.types.listOf lib.types.path;
+      default = [ ];
+      description = ''
+        Files to include into the config file using Taler's `@inline@` directive.
+
+        This allows including arbitrary INI files, including imperatively managed ones.
+      '';
+    };
+    configFile = lib.mkOption {
+      internal = true;
+      default =
+        let
+          includes = pkgs.writers.writeText "includes.conf" (
+            lib.concatStringsSep "\n" (map (include: "@inline@ ${include}") cfg.includes)
+          );
+          generatedConfig = settingsFormat.generate "generated-taler.conf" cfg.settings;
+        in
+        pkgs.runCommand "taler.conf" { } ''
+          cat ${includes} > $out
+          echo >> $out
+          echo >> $out
+          cat ${generatedConfig} >> $out
+        '';
+    };
```

Comment:
I think it would be better if this was taken out of `options` and passed to `common.nix` via other means, since it doesn't look like something the module user should be touching directly. Since you're already adding it to `environment.etc`, maybe you could reference this value in `common.nix` instead?

Reply:
This option defaults to the generated config file, but also allows the user to use an external one. Wouldn't this be useful?